Blog post on Feb LLM sessions (#451)
carpentries/carpentries.org: content/posts/2025/03/essential-knowledge-and-common-misconceptions.md

Participants at the sessions considered different aspects to be most important to teach in a workshop, including:

* It is still important to understand what your code does, even (especially\!) if an LLM generated parts of it for you.  
* How LLMs work: how they are trained, what training data was used, and how a response is chosen.  
* Different ways in which generative AI tools can be used to help with coding, e.g.   
  * asking for an explanation of what some code is doing or what an error message means.  
  * translating code between languages.  
  * describing what you want to do and having the LLM generate it for you.  

> * How to choose when an LLM is the appropriate tool for a given task, instead of searching the internet, for example, and why might you choose one tool over another.  

Reviewer: ```suggestion
* How to choose when an LLM is the appropriate tool for a given task, instead of searching the internet, for example, and why you might choose one tool over another.  
```

## Conclusion and Next Steps

Just like last time, the most immediate next step is another pair of community discussions scheduled for next week\! This time, we will learn from the experience of community members who have been teaching about and/or using LLMs in their own work. Some community members have been invited to present, but there will also be time for spontaneous contributions from participants and of course, for questions and discussion. These sessions will take place next Tuesday, 25 March, at [12:00 UTC](https://www.timeanddate.com/worldclock/fixedtime.html?msg=Carpentries+Community+Discussion&iso=20250325T1200) and [21:00 UTC](https://www.timeanddate.com/worldclock/fixedtime.html?msg=Carpentries+Community+Discussion&iso=20250325T2100) and [you can sign up to join these discussions on the community sessions Etherpad](https://pad.carpentries.org/community-sessions-2025).

> As discussed previously, the other next step is to update the existing curriculum to add some content discussing LLMs and their potential use. Based on the community discussions so far, informed by my own research and experience on the subject, I will prepare a pull request to make these changes to one of our lessons and invite feedback from community members within a defined timeframe. When those changes have been merged, I will call for help to roll out equivalent changes to the other lessons.

Reviewer: ```suggestion
As we previously discussed, the next step is to update the existing curriculum by incorporating content on LLMs and their potential applications. Based on the community discussions so far, informed by my own research and experience on the subject, I will prepare a pull request to make these changes to one of our lessons and invite feedback from community members within a defined timeframe. When those changes have been merged, I will call for help to roll out equivalent changes to the other lessons.
```
